File: uva online judge/1166 - Old Sorting.cpp
#include <cstdio>
#include <iostream>

using namespace std;
#define N 100

int main()
{
    int test,n,cs = 0, swaps;
    int in[N+1],pos[N+1];
    scanf("%d",&test);
    while(test--){
        scanf("%d",&n);
        for(int i = 1; i<=n; i++){
            scanf("%d",&in[i]);
            pos[in[i]] = i;
        }
        swaps = 0;
        for(int i = 1; i<=n; i++){
            if(in[i] != i){
                in[pos[i]] = in[i];
                pos[in[i]] = pos[i];
                in[i] = i;
                swaps++;
            }
        }
    printf("Case %d: %d\n", ++cs, swaps);
    }

    return 0;
}

File: uva online judge/1166 - Old Sorting.java
import java.util.Scanner;


public class Main {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		int test,n,swaps,cs = 0;
		int[] in,pos;
		Scanner sc = new Scanner(System.in);
		test = sc.nextInt();
		while((test--)>0){
			n = sc.nextInt();
			in =  new int[n+1];
			pos =  new int[n+1];
			
			for(int i = 1; i<=n; i++){
				in[i] = sc.nextInt();
				pos[in[i]] = i;
			}
			swaps = 0;
			for(int i = 1; i<=n; i++){
				if(in[i] != i){
					in[pos[i]] = in[i];
					pos[in[i]] = pos[i];
					in[i] = i;
					swaps++;
				}
			}
		System.out.printf("Case %d: %d\n", ++cs, swaps);
		}
	}

}
